lib/ansible/modules/remote_management/manageiq/manageiq_config.py

```diff
@@ -0,0 +1,226 @@
+    )
+
+    # seed the result dict in the object
+    result = dict(
+        changed=False,
+        name='',
+        original_value={},
+        value={},
+        diff={}
+    )
+
+    # get the origional value for the given config option
+    try:
+        original_value = get_manageiq_config_value(module, module.params['name'])
+    except Exception as err:
+        module.fail_json(msg=str(err), **result)
+
+    # update the result
+    result['name'] = module.params['name']
+    result['original_value'] = original_value
+
+    # create updated value dictionary
+    update_value = copy.deepcopy(original_value)
+    dict_merge(update_value, module.params['value'])
+
+    # enable diff mode
+    result['diff'] = {
+        'before': {
+            module.params['name']: original_value
+        },
+        'after': {
+            module.params['name']: update_value
+        }
+    }
+
+    # if check_mode then stop here
+    if module.check_mode:
+        result['changed'] = original_value != update_value
+        module.exit_json(**result)
+
+    # update config if difference
+    # else no-op
+    if original_value != update_value:
+        (update_value_rc, update_value_out, update_value_err) = module.run_command([
+            "rails",
+            "r",
+            "MiqServer.my_server.set_config(:%s => JSON.parse('%s'))" % (module.params['name'], json.dumps(update_value))
+        ], cwd=VMDB_PATH)
+        if update_value_rc != 0:
+            module.fail_json(msg="Error updating value for ':%s' config: %s" % (module.params['name'], update_value_err), **result)
+
+        # get the updated value for the given config option
+        try:
+            updated_value = get_manageiq_config_value(module, module.params['name'])
+        except Exception as err:
+            module.fail_json(msg=str(err), **result)
+    else:
+        updated_value = original_value
+
+    # setting the configuration is an async operation which must be waited for for completion
+    # NOTE: if anyone has a better idea how to do this please share
+    retries = 0
+    while update_value != updated_value and retries < CONFIRM_UPDATE_MAX_RETRY:
+        retries += 1
+        time.sleep(CONFIRM_UPDATE_SLEEP_INTERVRAL)
+        try:
+            updated_value = get_manageiq_config_value(module, module.params['name'])
+        except Exception as err:
+            module.fail_json(msg=str(err), **result)
+
+    # error if could not confirm update in timeout
+    if update_value != updated_value and retries >= CONFIRM_UPDATE_MAX_RETRY:
+        module.fail_json(
+            msg="Timed out waiting for set config to take affect. { 'retries': %s, 'sleep_interval': %s }" % (retries, CONFIRM_UPDATE_SLEEP_INTERVRAL),
+            **result
+        )
+
+    # verifty that the updated value equals the expected value
+    #     if the updated value does not match the expected value then error
+    #     else update or noop succesful
+    if update_value != updated_value:
```

Comment:
have you ever reached this `update_value != updated_value` ?

the while in L192 will run until `retries >= CONFIRM_UPDATE_MAX_RETRY` is true or `fail`
so if `update_value != updated_value` it must be also `retries >= CONFIRM_UPDATE_MAX_RETRY` and
then you will `fail` in L202 ?

am I missing something ?